src/core/varinfo.jl

```diff
@@ -92,10 +96,21 @@ setgid!(vi::VarInfo, gid::Int, vn::VarName) = vi.gids[getidx(vi, vn)] = gid
 istrans(vi::VarInfo, vn::VarName) = vi.trans[end][getidx(vi, vn)]
 settrans!(vi::VarInfo, trans::Bool, vn::VarName) = vi.trans[end][getidx(vi, vn)] = trans
 
-getlogp(vi::VarInfo) = vi.logp[end]
-setlogp!(vi::VarInfo, logp::Real) = vi.logp[end] = logp
-acclogp!(vi::VarInfo, logp::Real) = vi.logp[end] += logp
-resetlogp!(vi::VarInfo) = setlogp!(vi, zero(Real))
+getlogprior(vi::VarInfo) = vi.logprior[end]
+setlogprior!(vi::VarInfo, logp::Real) = vi.logprior[end] = logp
+acclogprior!(vi::VarInfo, logp::Real) = vi.logprior[end] += logp
+resetlogprior!(vi::VarInfo) = setlogprior!(vi, zero(Real))
```

Comment:
Seems too many levels of interfaces here: as `resetlogprior!` only used in `resetlogp!`, maybe it is not necessary.